plugins/migration/bin/rename.js

```diff
@@ -0,0 +1,286 @@
+ * @deprecated Use Renamer instance instead.
+ */
+export function doRenamings(database, currVersion, newVersion, strings) {
+  const renamer = new Renamer(database, currVersion, newVersion);
+  return strings.map((str) => renamer.rename(str));
+}
+
+/**
+ * An object which can apply a given set of renamings to the contents
+ * of JavaScript files, so as to automatically update developer's code
+ * where parts of the Blockly API have been renamed.
+ */
+export class Renamer {
+  /**
+   * Constructor for the class.
+   * @param {!Object} database The database of renames.
+   * @param {string} currVersion The version to migrate from.
+   * @param {string} newVersion The version to migrate to.
+   */
+  constructor(database, currVersion, newVersion) {
+    /** @private @const {!Array<!VersionRenamer>} */
+    this.versionRenamers_ =
+        Renamer.calculateRenamings(database, currVersion, newVersion);
+  }
+
+  /**
+   * Turns a database of renamings into an intermediate format that is
+   * easier to apply.
+   * @param {!Object} database The database of renames.
+   * @param {string} currVersion The version to migrate from.
+   * @param {string} newVersion The version to migrate to.
+   * @return {!Array<!VersionRenamer>} The collection of renamings to perform.
+   */
+  static calculateRenamings(database, currVersion, newVersion) {
+    currVersion = Renamer.coerce(currVersion);
+    newVersion = Renamer.coerce(newVersion);
+    const versions = Object.keys(database).sort(Renamer.compare);
+    const renamers /** !Array<!VersionRenamer> */ = [];
+    for (const version of versions) {
+      // Only process versions in the range (currVersion, ^newVersion].
+      if (Renamer.lte(version, currVersion)) continue;
+      if (Renamer.gt(version, newVersion)) break;
+
+      renamers.push(new VersionRenamer(database[version]));
+    }
+    return renamers;
+  }
+
+  /**
+   * Coerces the given string into a valid version (semver compliant or
+   * develop).
+   * @param {string} version  The version to coerce.
+   * @return  {string} The coerced version.
+   */
+  static coerce(version) {
+    return version === DEV_VERSION ?
+        version : semver.coerce(version).toString();
+  }
+
+  /**
+   * Compares the given versions. Compatible with Array.sort.
+   * @param {string} v1 The first version to compare.
+   * @param {string} v2 The second version to compare.
+   * @return {number} A number indicating the relationship between the versions.
+   */
+  static compare(v1, v2) {
+    if (v2 === DEV_VERSION) return -1;
+    if (v1 === DEV_VERSION) return 1;
+    return semver.compare(v1, v2);
+  }
+
+  /**
+   * Returns true if the first version is less than or equal to the second
+   * version.
+   * @param {string} v1 The version to compare.
+   * @param {string} v2 The version to compare against.
+   * @return {boolean} True if the first version is less than or equal to the
+   *     second one.
+   */
+  static lte(v1, v2) {
```

Comment:
I'd prefer a more descriptive name (and for `gt`), but I can't actually think of anything I like better.  :-/

Reply:
Made these descriptive by moving them to a new versions file. The renamer now uses `versionUtils.lte` which I think is clear enough.